lib/application/widgets/unfocus_detector.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/material.dart';
+
+/// A wrapper Widget that request unfocus when tapping outside [child].
+class UnfocusDetector extends StatelessWidget {
```

Comment:
I'm not sure if I follow the usage of this widget. Could you explain why this is needed? If it's an "unfocus" detector, how come it doesn't have a "onUnfocus"? Also, wouldn't it conflict with other "focus" elements in the `child` element?

If, for example, we have multiple children in this `child` and one of them is focused, won't tapping on another one that requests focus, unfocusing the one existing and focus on the next one? I know it may sound like the obvious, but this shouldn't be done by us, instead, it should be handled by Flutter and can potentially lead to "visual inconsistencies".

Why don't we handle this "unfocusing" as per widget-basis and let each implementation handle when it should unfocus its context?

Reply:
> Could you explain why this is needed?

I think this is an easy to use and must-have when we have any screen that the keyboard may appear, since it's possible to hide the keyboard tapping anywhere that doesn't have a TextField.

> If it's an "unfocus" detector, how come it doesn't have a "onUnfocus"?

Because we may use from Flutter singleton `FocusManager.instance`. The idea is not to expose this control to the parent Widget do something, it's just to hide the keyboard by removing the focus from the focused Widget (if it exists).

> Also, wouldn't it conflict with other "focus" elements in the child element?

It doesn't. I've made a few tests and Widgets like `TextField` defer the gestures to its child, which mean that this Widget gesture detector will not be used when tapping intractable elements. We may add some unit tests for these scenarios (TextField, buttons, scrollviews, etc.), although I think it's not necessary, since we already used Widgets like this one in other projects and we haven't faced problems.

> Why don't we handle this "unfocusing" as per widget-basis and let each implementation handle when it should unfocus its context?

As I mentioned above, the goal is not to provide a custom behavior to the parent Widget, since the use-case here will be always the same: The user want to dismiss the keyboard (i.e, unfocus some TextFiled) by tapping in non-intractable elements).

Reply:
Then I would rename it, because if I'm not mistaken, Flutter uses `Detector` as a passive widget that provides gesture/behaviors detection. Maybe `AlwaysUnfocus`?

And also, I would explain when it should unfocus, as the docs seem to not address these. Things like: if the child doesn't respond for gestures, it will trigger the unfocus, otherwise it won't.

This widget could be more tested IMO, as it has a really significant responsibility and a lot of possible use-cases. My suggestion on these tests would be with a `child` that DO handle gestures and another that DO NOT handle. Could also possibly test with fields, those that trigger the keyboard appearance.

Reply:
> Flutter uses Detector as a passive widget that provides gesture/behaviors detection

But isn't that what we are aiming to do here? I mean, detect the act of unfocusing a field based on a tap gesture. But I'm ok with changing the name if you prefer.

> And also, I would explain when it should unfocus, as the docs seem to not address these. Things like: if the child doesn't respond for gestures, it will trigger the unfocus, otherwise it won't.

Updated the docs.

> This widget could be more tested IMO

I was writing more tests to the Widget when I realized that the unique test cover all the cases that we expect. If you take a look at it, it covers both scenarios: Interacting with a focusable Widget (TextFiled) and with a non-focusable Widget (Container) and verify if the focus is managed accordingly. I was thinking about writing tests with buttons, scrollviews and others but it doesn't make sense because they are not focusable. The usage of this usage implies wrapping TextFields and other Widgets that receive the user input, and these are covered by this unique unit test.

Reply:
> But isn't that what we are aiming to do here? I mean, detect the act of unfocusing a field based on a tap gesture. But I'm ok with changing the name if you prefer.

Yes, that's what we would do if we would want to detect something, but not to take actions upon our behalf. It doesn't only detect but also acts upon.

That is my perspective on why it would make sense to not use the same naming as `Detector`, but that's up to you honestly, just my 2 cents, maybe like `AbsorbPointer` or `IgnorePointer`, we could use `UnfocusPointer`?

Reply:
I liked `UnfocusPointer`. Renamed the Widget